### PR TITLE
Erijpkema made example work

### PR DESCRIPTION
**Pull request by erijpkema**
Some changes I found usefull when using cacafonisk for our internal project. Each commit message hopefully explains my reasons.

### CODE REVIEW

**Comment by JorisE** [comment]
Comments ended up in: https://github.com/VoIPGRID/cacofonisk/pull/6

**Comment by erijpkema** [comment]
Am a bit dissapointed in my vim !sort. Will fix that ;-)

File: README.rst [hunk 1]
```diff
@@ -246,3 +248,33 @@ The CallerId contains the following information about participants in a call:
 
 The CallerId is passed to the ``on_b_dial`` and ``on_transfer`` methods of a
 reporter.
+
+Write tests
+-----------
+
+You can write a testcase from a json eventlog. below is an example that does not test anything. You could replace the reporter or the chanmgr with a version of your own that needs testing.
```
Comment (JorisE) [inline README.rst]: s/chanmgr/channel_manager

File: README.rst [hunk 2]
```diff
@@ -246,3 +248,33 @@ The CallerId contains the following information about participants in a call:
...
+    from cacofonisk.channel import CallerId, ChannelManager
+
+    class MyVeryOwnTestCase(BaseTestCase):
+        """
+        Tests my own code.
+        """
+
+        def test_on_notify_callstate_cloudcti(self):
+            """
+            Play a log and test *stuff*
```
Comment (JorisE) [inline README.rst]: line ending.

File: cacofonisk/utils/testcases.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+class BaseTestCase(TestCase):
+    """
+    BaseTestCase that has methods to open and parse json event logs.
+    """
+    def open_file(self, filename, *args, **kwargs):
+        """"
+        Open a filename. Aditional arguments are passed to open().
+
+        Args:
+            filename (str) File to open (relative to current path)
```
Comment (JorisE) [inline cacofonisk/utils/testcases.py]: (str):

and line ending.

File: cacofonisk/utils/testcases.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        path = os.path.dirname(__file__)
+        filename = os.path.join(path, filename)
+        return open(filename, *args, **kwargs)
+
+    def load_events_from_disk(self, filename):
+        """
+        Load events from a json event log.
+
+        Args:
+            filename (str) File to open (relative to current path).
```
Comment (JorisE) [inline cacofonisk/utils/testcases.py]: (str):

**Comment by JorisE** [comment]
We might have to review our review process for github :). I may have lost sight of the woods in between the trees.

If you could squash the commits, I'm pretty sure I'd gladly merge this and do a new release.

**Pull request closed without merging**